### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ceyuen**
@phylix

### CODE REVIEW

File: Add Py to string.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*21. Write a JavaScript program to create a new string adding "Py" in front of a given string. If the given string begins with "Py" then return the original string. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var string = prompt('Enter a phrase or word'); 
+//var string = 'Pyche'; 
+
+if (string.startsWith('Py')){
+    console.log(string);
```
Comment (phylix) [inline Add Py to string.js]: We can put this in a function that accepts a string as a parameter and returns the output string.  We can then use `console.log()` to print the output string.

This gives us more flexibility in how we want to output the final string.  Such as logging it to console, writing it to a file, passing it to another function, etc.

File: Add first character of string.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* 24. Write a JavaScript program to create a new string from a given string with the first character of the given string added at the front and back. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+//var str = prompt('Enter a word or phrase');
+var str = 'hi';
+
+var firstStr = str.charAt(0);
+
+console.log(firstStr + str + firstStr);
```
Comment (phylix) [inline Add first character of string.js]: Lets make this a function, too.

File: Check array.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php*/
+
+function check(numbers){
+    return numbers.indexOf(1) === -1 && numbers.indexOf(3) === -1;
+}
+
+console.log(check([1, 2, 5]));
+console.log(check([4,2,3]));
+console.log(check([1,3,5]));
+console.log(check([4,5,6]));
```
Comment (phylix) [inline Check array.js]: The instructions say the length of the array is 2, so that's what should be tested.  However, I do really like that your solution works for arrays of any size.  Good job.

File: Check array.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/*78. Write a JavaScript program to test if an array of integers of length 2 does not contain 1 or a 3.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php*/
+
+function check(numbers){
```
Comment (phylix) [inline Check array.js]: A more descriptive method name will be good practice when you start to write larger programs.  Lets get you in the habit of doing that by using descriptive method names here.

File: Check beginning letters.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/* 66. Write a JavaScript program to display the city name if the string begins with "Los" or "New" otherwise return blank.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function city(string){
```
Comment (phylix) [inline Check beginning letters.js]: Lets change this method name to something more descriptive for practice.

File: Check first and last of array.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/* 72. Write a JavaScript program to check if the first and last elements are equal of a given array of integers length 3.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function check(numbers){
```
Comment (phylix) [inline Check first and last of array.js]: Lets change this method name to something more descriptive for practice.

File: Check first and last of array.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+    return numbers[0] === numbers[numbers.length-1];
+}
+
+console.log(check([100, 293, 423, 100]));
```
Comment (phylix) [inline Check first and last of array.js]: Again, good job on the solution being able to handle arrays of any size.

It looks like you may have length and largest index confused.  The length of the array is the number of elements in the array and the index of the last element is `length - 1`.  So for this problem, the arrays being passed in should have only 3 elements and the last element should have an index of 2.  

Example: `[100, 293, 424]' has a length of 3 and the largest index is 2.

File: Check for Java.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+/*27. Write a JavaScript program to check if a string starts with 'Java' and false otherwise.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var string;
```
Comment (phylix) [inline Check for Java.js]: We don't need to declare this variable since it's not used.

File: Check for Java.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+function checkString(string){
+    return (string.startsWith('Java'));
```
Comment (phylix) [inline Check for Java.js]: We don't need to have the extra parentheses around `string.startsWith('Java')`

File: Check last digits.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+/* 36. Write a JavaScript program to check if the last digit of the three given positive integers is same.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var ints = [866,96,26];
+var newInts = [];
```
Comment (phylix) [inline Check last digits.js]: We can assign `newInts` the value of the `map()` method from below to simplify the code a tiny bit.

Comment (phylix) [reply Check last digits.js]: We should also move the mapping code inside the method since it's created specifically for the method.  This will prevent a ton of variables polluting the rest of your code.  It can make it difficult to change the code in the future if those variables are being used elsewhere in the code because you'll have to check everything and then account for the removal of the variable in other parts of the code.

File: Check palindrome.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
+/*2. Write a JavaScript function that checks whether a passed string is palindrome or not? 
+A palindrome is word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.
+
+https://www.w3resource.com/javascript-exercises/javascript-functions-exercises.php*/
+
+function palindrome(string){
```
Comment (phylix) [inline Check palindrome.js]: We should avoid using types as variable names to avoid potential confusion.

File: Check palindrome.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    var splitString = string.split("");
```
Comment (phylix) [inline Check palindrome.js]: Does `split()` do anything when you pass in an empty string?

Also, why do we need to split the string?

File: Check palindrome.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    var revString = splitString.reverse();
+    var newString = revString.join("");
+    return newString === string ? true : false
```
Comment (phylix) [inline Check palindrome.js]: This would incorrectly return false for "nurses run"

File: Check range.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+//var num = prompt('Enter a number');
+var num; 
+
+function checkRange(num) { 
+    return num <= 60 && num >= 40 || 70 <= num  && num <= 100; 
+}
+
+console.log(checkRange(150));
```
Comment (phylix) [inline Check range.js]: Lets add some more examples that test different values to show the different possibilities.  Something less than 40, something between 40 and 60, something between 61 and 69, something between 70 and 100, and something over 100.

File: Check sum for 15.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+var x, y; 
+
+function checkFifteen(x, y){
+    return (x === 15 || y === 15 || x + y === 15 || Math.abs(x - y) === 15);
+}
+
+console.log(checkFifteen(15,23));
+console.log(checkFifteen(5,10));
+console.log(checkFifteen(20,5));
+console.log(checkFifteen(8,8));
```
Comment (phylix) [inline Check sum for 15.js]: Lets add more test cases to test each requirement.  It looks like you're missing the case where `y = 15` and when `x - y = -15`

File: Convert number to hours and minutes.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+/* 51. Write a JavaScript program to convert a given number to hours and minutes. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var num, hour, min; 
+
+function convertTime(num){
+    hour = Math.floor(num / 60)
+    min = num % 60
```
Comment (phylix) [inline Convert number to hours and minutes.js]: Lets add semi-colons for these lines for consistency.

File: Convert number to hours and minutes.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
+/* 51. Write a JavaScript program to convert a given number to hours and minutes. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var num, hour, min;
```
Comment (phylix) [inline Convert number to hours and minutes.js]: We should declare these variables inside the function.

File: Convert number to hours and minutes.js [hunk 18]
```diff
@@ -0,0 +1,13 @@
+/* 51. Write a JavaScript program to convert a given number to hours and minutes. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var num, hour, min; 
+
+function convertTime(num){
+    hour = Math.floor(num / 60)
+    min = num % 60
+    console.log(hour + " hours and " + min + " minutes");
```
Comment (phylix) [inline Convert number to hours and minutes.js]: Instead of logging, lets return the string.

File: Copy strings.js [hunk 19]
```diff
@@ -0,0 +1,15 @@
+/* 57. Write a JavaScript program to create a new string of specified copies (positive number) of a given string.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function copy(string, int){
```
Comment (phylix) [inline Copy strings.js]: We should avoid using datatypes as variable names to avoid confusion.

File: Days left until Christmas.js [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+if (today.getMonth() === 11 && today.getDate() === 25){
+    console.log('Ho! Ho! Ho! Today is Christmas! Merry Christmas!')
+}else if (today.getMonth() === 11 && today.getDate() > 25) { 
+    var xmas = new Date (today.getFullYear() + 1, 11, 25); 
+    var daysLeft = [ xmas.getTime() - today.getTime() ] / 86400000;
+    console.log('Christmas passed this year already, but ' + daysLeft + ' days until next Christmas!')
+}else {
+    var daysLeft = [ xmas.getTime() - today.getTime() ] / 86400000;
+    console.log(Math.ceil(daysLeft) + " days left until Christmas!");
+}
```
Comment (phylix) [inline Days left until Christmas.js]: Lets turn this into a function.

File: Days left until Christmas.js [hunk 21]
```diff
@@ -0,0 +1,16 @@
+/*Excercise 9 from https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php
+
+Write a JavaScript program to calculate days left until next Christmas.*/
+
+var today = new Date();
+var xmas = new Date (today.getFullYear(), 11, 25); 
+if (today.getMonth() === 11 && today.getDate() === 25){
+    console.log('Ho! Ho! Ho! Today is Christmas! Merry Christmas!')
+}else if (today.getMonth() === 11 && today.getDate() > 25) { 
+    var xmas = new Date (today.getFullYear() + 1, 11, 25);
```
Comment (phylix) [inline Days left until Christmas.js]: `xmas` variable is declared in this code block and outside.  It can get confusing which when you're using if the code is longer.  Also, if you want to use the `xmas` variable that was declared earlier, you wouldn't be able to.  I'm surprised this is allowed.  It usually isn't in other languages.

File: Determine leap year.js [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+
+if (checkYear % 4){
+    console.log(checkYear + " is not a leap year.");
+}else if (checkYear % 100){ 
+    console.log(checkYear + " is a leap year.");
+}else if (checkYear % 400){
+    console.log(checkYear + " is not a leap year.");
+}else {
+        console.log(checkYear + " is a leap year.");
+}
```
Comment (phylix) [inline Determine leap year.js]: Lets change this into a function.

File: Difference between number and 13.js [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+var difference;
+if (isNaN(number)){
+    console.log('Enter a number')
+} else if (number <= 13) {
+    difference = (13 - number);
+    console.log(difference + " is the absolute difference between " + number + " and 13");
+}else {
+    difference = 2 * (number - 13);
+    console.log(difference + " is double the absolute difference between " + number + " and 13");
+}
```
Comment (phylix) [inline Difference between number and 13.js]: Lets turn this into a function.

File: Check first and last of array.js (hunk 7 above)
Comment (ceyuen) [reply Check first and last of array.js]: Problem said integers of length 3. I assumed it meant numbers with 3 digits.

File: Excercise 4.1 from Beginning Javascript.js [hunk 24]
```diff
@@ -0,0 +1,27 @@
+/* Excercise 4.1 from Beginning Javascript by McPeak and Wilton
+
+Change the code of question 2 from Chapter 3 so that it's a function that takes as parameters the times table required and hte vlaues at which it should start and end. FOr example, ou might try the 4 times table displayed start with 4*4 and ending at 4*9*/
+
+
+function twelveTimes(multiplier, startNum, endNum){
```
Comment (phylix) [inline Excercise 4.1 from Beginning Javascript.js]: This function name can be improved.  It doesn't accurately describe what it's doing since the multiplier is a parameter instead of always being 12.

File: Check beginning letters.js [hunk 25]
```diff
@@ -0,0 +1,14 @@
+/* 66. Write a JavaScript program to display the city name if the string begins with "Los" or "New" otherwise return blank.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function startsWithLosOrNew(cityName){
+    if (cityName.startsWith('Los') || cityName.startsWith('New')){
+        return cityName;
+    } else { 
+        return ' '; 
+    }
```
Comment (phylix) [inline Check beginning letters.js]: This is perfectly fine.  An alternative is:

```
function startsWithLosOrNew(cityName) {
  return cityName.startsWith('Los') || cityName.startsWith('New') ? cityName : ' ';
}
```

File: Check beginning letters.js [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+console.log(startsWithLosOrNew('New York'));
+console.log(startsWithLosOrNew('Seattle'));
```
Comment (phylix) [inline Check beginning letters.js]: It'd be good if you added an example for 'Los'

File: Check first and last of array.js [hunk 27]
```diff
@@ -0,0 +1,9 @@
+/* 72. Write a JavaScript program to check if the first and last elements are equal of a given array of integers length 3.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function firstLastElementsEqual(numberArr){
+    return numberArr[0] === numberArr[numberArr.length-1];
+}
+
+console.log(firstLastElementsEqual([100, 293, 423, 100]));
```
Comment (phylix) [inline Check first and last of array.js]: It'd be good if you had an example for when the first and last elements of the array are not equal, too.

File: Check for 50.js [hunk 28]
```diff
@@ -0,0 +1,9 @@
+/* 18. Write a JavaScript program to check two given numbers and return true if one of the number is 50 or if their sum is 50. 
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function isFifty(n1, n2) { 
+    return n1 === 50 || n2 === 50 || n1 + n2 === 50; 
+}
+
+console.log(isFifty(3,50));
```
Comment (phylix) [inline Check for 50.js]: Lets add some tests where 50 is the sum and 50 is neither of the numbers nor the sum.

File: Check for Java.js [hunk 29]
```diff
@@ -0,0 +1,10 @@
+/*27. Write a JavaScript program to check if a string starts with 'Java' and false otherwise.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function startWithJava(input){
+    return input.startsWith('Java');
```
Comment (phylix) [inline Check for Java.js]: Lets start checking the input for nulls to prevent exceptions (errors).  You just need to do `if(input == null)` then return something appropriate.  In this case, maybe return null.

Also, keep in mind that sometimes it's appropriate to throw an exception.

File: Check for Java.js [hunk 30]
```diff
@@ -0,0 +1,10 @@
...
+}
+
+console.log(startWithJava('Javascript'));
+console.log(startWithJava('Hello'));
```
Comment (phylix) [inline Check for Java.js]: Lets add a test case where the string we give the function is null.

File: Check for Script.js [hunk 31]
```diff
@@ -0,0 +1,12 @@
...
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function checkString(string){
+    return string.substring(4,10) === 'Script' 
+    ? string.slice(0,4) + string.slice(10, string.length) 
+    : string;
+    
+}
+
+console.log(checkString('JavaScriptsdf'));
```
Comment (phylix) [inline Check for Script.js]: Lets add a test without "Script" in the 5th position and one without "Script" in the string at all.  And one with null as the input.

File: Check last digits.js [hunk 32]
```diff
@@ -0,0 +1,11 @@
+/* 36. Write a JavaScript program to check if the last digit of the three given positive integers is same.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function checkDigits(arr){
+    var newInts = arr.map(x => x % 10);
```
Comment (phylix) [inline Check last digits.js]: Check nulls

File: Check palindrome.js [hunk 33]
```diff
@@ -0,0 +1,53 @@
+/*2. Write a JavaScript function that checks whether a passed string is palindrome or not? 
+A palindrome is word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.
+
+https://www.w3resource.com/javascript-exercises/javascript-functions-exercises.php*/
+
+function palindrome(input){
+	var removeSpaces = input.split(' ').join('');
+    return removeSpaces.split('').reverse().join('') === removeSpaces ? true : false
```
Comment (phylix) [inline Check palindrome.js]: This can just be `return removeSpaces.split('').reverse().join('') === removeSpaces;`

File: Check palindrome.js [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+
+console.log(palindrome('madam'));
+console.log(palindrome('nurses run'));
```
Comment (phylix) [inline Check palindrome.js]: Lets add examples of phrases that aren't palindromes.

File: Check palindrome.js (hunk 33 above)
Comment (phylix) [reply Check palindrome.js]: I like that your solution is much shorter and concise than the given solution.  Good job.

File: Check palindrome.js [hunk 35]
```diff
@@ -0,0 +1,53 @@
+/*2. Write a JavaScript function that checks whether a passed string is palindrome or not? 
+A palindrome is word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.
+
+https://www.w3resource.com/javascript-exercises/javascript-functions-exercises.php*/
+
+function palindrome(input){
+	var removeSpaces = input.split(' ').join('');
```
Comment (phylix) [inline Check palindrome.js]: This is fine, but just to make sure you know that there are also other ways to remove spaces.

`input.replace(/ /g,'')` is more straightforward to me when reading it.

https://stackoverflow.com/questions/6623231/remove-all-white-spaces-from-text

File: Check sum for 15.js [hunk 36]
```diff
@@ -0,0 +1,14 @@
...
+console.log(checkFifteen(0,15));
```
Comment (phylix) [inline Check sum for 15.js]: This test satisfies both cases I mentioned (`y = 15` and `x - y = 15`), but it's better to test them separately.

If there were a bug where everything works except for `x - y = 15`, then we would think that everything is fine.  WIth `checkFifteen(0, 15)`, it would pass because it would immediately return true when it reaches `y === 15`. 

If your code were changed to have a bug, all of these test cases would work and can be misled into thinking everything was fine.  Example:

```
function checkFifteen(x, y){
     return (x === 15 || y === 15 || x + y === 15 || x - y === 15);
 }
```

It's very good that you thought of using `Math.abs()` because it can be easily missed.

File: Excercise 4.1 from Beginning Javascript.js [hunk 37]
```diff
@@ -0,0 +1,27 @@
...
+    return timesTable; 
+
+}
+
+console.log(twelveTimes(4,4,9));
+
+/* Answer in book: 
+
+function writeTimesTable(timesTable, timesByStart, timesByEnd){
+for (; timesByStart < = timesByEnd ; timesByStart++){
```
Comment (phylix) [inline Excercise 4.1 from Beginning Javascript.js]: Something to take note of from the book's solution is that you don't have to fill in each part of a `for` loop.  They've left the first part blank and it's legal and it's the same for any of the other three parts.

`for([Executed pre-Loop]; [Pre-loop start condition]; [Executed post-loop])`

With pre-loop and post-loop execution parts empty, it's like a while loop.  
With pre-loop empty, it's like a while loop with post-loop added to the bottom of the code block. 
With post-loop empty, it's like a while loop with the pre-loop code outside of the while loop, right above it.
With the start condition missing, it's an infinite loop that is like `while(true)` and you need a `break;` statement to end the loop.

File: Find PI to the Nth Digit.js [hunk 38]
```diff
@@ -1,11 +1,12 @@
...
+   var pieString = Math.PI.toString(); 
    
-   if (digitPlace > 30 || digitPlace <=0){
+   if (digitPlace > 30 || digitPlace <=0) {
       console.log("Select a digit from 1-30");
-  }else { console.log(pieString.slice(0,digitPlace+2));}
\ No newline at end of file
+    } else { 
+      console.log(pieString.slice(0,digitPlace+2));
+    }
```
Comment (phylix) [inline Find PI to the Nth Digit.js]: Lets turn this into a function and have different tests

File: Head First_Bubbles function.js [hunk 39]
```diff
@@ -0,0 +1,62 @@
...
+    for (var i = 0; i < arr.length; i++){
+        str += 'Bubble solution #' + i + ' score: ' + arr[i] + '\n';
+    }
+    return str; 
+}
+
+function findHighScore(arr){
+    var highScoreValue = 0;
+    for (var i = 0; i < arr.length; i++){
+        if (highScoreValue <= arr[i+1]){
```
Comment (phylix) [inline Head First_Bubbles function.js]: If `highScoreValue === arr[i+1]`, do we still want to update the value of `highScoreValue`?

File: Head First_Bubbles function.js [hunk 40]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+    return indexOfHighScore;
+}
+
+function cost(arr, costArr){
+    var indexOfHighScore = findHighScoreIndex(arr);
+    for (var k = 0; k < indexOfHighScore.length; k++){
+        if (costArr[indexOfHighScore[k]] < costArr[indexOfHighScore[k+1]]){
+            var indexOfLowestCost = indexOfHighScore[k];
```
Comment (phylix) [inline Head First_Bubbles function.js]: Declaring the variable inside the if block and then using it outside works for JavaScript, but not for most other languages.  This is a very good thing to know and keep in mind when you start programming in other languages.

Read this:
https://edgecoders.com/function-scopes-and-block-scopes-in-javascript-25bbd7f293d7

File: Head First_Bubbles function.js [hunk 41]
```diff
@@ -0,0 +1,62 @@
...
+            indexOfHighScore.push(j);
+        }
+    }
+    return indexOfHighScore;
+}
+
+function cost(arr, costArr){
+    var indexOfHighScore = findHighScoreIndex(arr);
+    for (var k = 0; k < indexOfHighScore.length; k++){
+        if (costArr[indexOfHighScore[k]] < costArr[indexOfHighScore[k+1]]){
```
Comment (phylix) [inline Head First_Bubbles function.js]: Won't `k+1` eventually reach an index that is outside of the range of `indexOfHighScore`?  

Example: `indexOfHighScore = [2]`, so `indexOfHighScore.length` is 1.

On the first loop iteration, it will try `indexOfHighScore[k+1]`, which is `indexOfHighScore[1]`, which is out of range.

File: New array.js [hunk 42]
```diff
@@ -0,0 +1,12 @@
+/*75. Write a JavaScript program to create a new array taking the middle elements of the two arrays of integer and each length 3.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php*/
+
+var ne;
```
Comment (phylix) [inline New array.js]: We should declare this variable inside the function

File: Remove first and last letter.js [hunk 43]
```diff
@@ -0,0 +1,9 @@
+/* 60. Write a JavaScript program to create a new string without the first and last character of a given string.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+function remove(string){
+    return string.slice(1, string.length-1);
```
Comment (phylix) [inline Remove first and last letter.js]: Lets also think about when the string being passed in is empty, null, and length of 1.  This would be a good basic interview question that tests how aware you are of edge cases.  In this case, the interesting situation is when the string is a single character.

File: Reverse string.js [hunk 44]
```diff
@@ -0,0 +1,11 @@
...
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+var string = 'hello'; 
+
+var splitString = string.split("");
+var revString = splitString.reverse();
+var newString = revString.join("");
+
+console.log(newString);
```
Comment (phylix) [inline Reverse string.js]: Lets make this a function

File: Sum of 2 numbers.js [hunk 45]
```diff
@@ -0,0 +1,18 @@
+/* 39. Write a JavaScript program to compute the sum of the two given integers, If the sum is in the range 50..80 return 65 other wise return 80.
+
+https://www.w3resource.com/javascript-exercises/javascript-basic-exercises.php */
+
+//var x = prompt ('Enter first number');
+//var y = prompt ('Enter second number');
+var x, y;
```
Comment (phylix) [inline Sum of 2 numbers.js]: We should declare these inside the function

File: Sum of 2 numbers.js [hunk 46]
```diff
@@ -0,0 +1,18 @@
...
+function sum(x, y){
+    if (x + y > 50 && x + y < 80){
+        return 65;
+    } else {
+        return 80;
+    }
+}
+
+console.log(sum(5,70));
+console.log(sum(5,5));
```
Comment (phylix) [inline Sum of 2 numbers.js]: Should have a test where the sum is greater than 80, too
